.gitlab-ci.yml

```diff
@@ -0,0 +1,68 @@
+include:
+  - project: hpc/gitlab-pipelines
+    file: spack-build-components.gitlab-ci.yml
```

Comment:
q : when some some changes these common template file, are dependent CI projects are triggered so that there is no regression?

Reply:
Good question. It doesn't happen automatically, but @matz-e already set up the CI of `hpc/gitlab-pipelines` to trigger the CI of `hpc/touchdetector` as a regression test. Once this PR is merged I think we should add CoreNEURON's CI alongside TouchDetector's to improve the test coverage.